Release v5.0.0

illusionfield opened the pull request:
### Summary

This PR introduces `fourseven:scss@5.0.0-beta.1`, which completes the migration to Dart Sass and officially removes the built-in `sass` dependency.

It combines the changes from two previous PRs into one consolidated update targeting the `release/5.0.0` branch.

---

### What's Changed

- 💥 **Breaking**: `sass` is no longer bundled — users must now install either `sass` or `sass-embedded` manually via `npm`
- 🚀 **Flexibility**: Developers can now pin versions or switch Sass implementations without modifying this plugin
- 📚 **Docs**:
  - README has been restructured and clarified to reflect the new usage model
  - Migration Guide expanded and updated, now includes:
    - Manual Sass installation
    - Sass engine flexibility
    - PostCSS setup
    - Modern `@use`/`@forward` transition instructions
- 🧼 Removed legacy references to `node-sass` and outdated patterns
- 🧪 Debug mode and custom config options remain fully supported

---

### Migration Notes

To upgrade to this version, you **must** manually install a Sass compiler:

```bash
meteor npm install --save-dev sass
# or
meteor npm install --save-dev sass-embedded
```

For more details, see the [Migration Guide](https://github.com/illusionfield/fourseven-scss/blob/master/MIGRATION-GUIDE-v5.md)

---

### Why This Matters

- Keeps the package leaner and less opinionated
- Unlocks performance with `sass-embedded` (native binary)
- Makes the plugin more robust for future Meteor 3 releases
- Easier for community contributors to reason about dependencies

---

### Resolved Issues

- Closes #327
- Closes #324

---

This replaces and supersedes the two earlier PRs (now closed):
- [PR #323] (original Dart Sass rewrite)
- [PR #328] (remove direct sass dependency)

illusionfield committed:
remove @babel/runtime dependency

https://github.com/Meteor-Community-Packages/meteor-scss/pull/323#discussion_r1944648741

illusionfield committed:
Migration Guide

illusionfield committed:
Internal Meteor Packages: improve root file detection logic for imports and package files
https://github.com/Meteor-Community-Packages/meteor-scss/issues/324

illusionfield committed:
Implement includePaths option
https://github.com/Meteor-Community-Packages/meteor-scss/pull/323#issuecomment-2659045373

illusionfield committed:
Merge branch 'master' into develop

illusionfield committed:
Merge branch 'hotfix/internal-packages'

illusionfield committed:
Merge branch 'master' into feature/documentation

* master:
  Internal Meteor Packages: improve root file detection logic for imports and package files https://github.com/Meteor-Community-Packages/meteor-scss/issues/324

illusionfield committed:
readme draft

illusionfield committed:
Merge branch 'feature/documentation' into develop

jankapunkt committed:
published 5.0.0-alpha.2

illusionfield committed:
fix #324
fully tested!

illusionfield committed:
version bump

illusionfield committed:
Merge branch 'dat-css'

illusionfield committed:
windows compatibility fix

https://github.com/Meteor-Community-Packages/meteor-scss/issues/327

illusionfield committed:
decode URI components in loaded URLs for correct file path handling

https://github.com/Meteor-Community-Packages/meteor-scss/issues/327

illusionfield committed:
include paths `Scss compiler Error: Invalid URL:` bugfix

https://github.com/Meteor-Community-Packages/meteor-scss/issues/324

illusionfield committed:
Update documentation

https://github.com/Meteor-Community-Packages/meteor-scss/pull/323#pullrequestreview-2621354448
https://github.com/Meteor-Community-Packages/meteor-scss/pull/323/files/139f02f0023293b2a045c96d6217e28ab2dcd8bd

illusionfield committed:
update SASS to v1.85.1

illusionfield committed:
Add GitHub Actions test suite

illusionfield committed:
Bump version to 5.0.0-beta.1

illusionfield committed:
Migration Guide typos correction

illusionfield committed:
ci(test): add support for Meteor 2.10 in GitHub Actions matrix

illusionfield committed:
fix: resolve npm dependency installation issue in Meteor 2 package tests

Addressed an issue where running `meteor test-packages` in Meteor 2 failed due to npm dependency installation errors related to the `meteor-node-stubs` update. This fix ensures compatibility and resolves the 'npmignore: command not found' error during package tests.

For more details, see: https://github.com/meteor/meteor/issues/13691

illusionfield committed:
feat!: decouple Sass version management from plugin (v5.1.0-beta.1)

- Removed `sass` as a direct dependency to allow full control over Sass versioning
- Enables usage of both `sass` and `sass-embedded` via devDependencies
- Updated README.md for improved clarity, examples, and migration guidance
- Bumped version to `5.1.0-beta.1` to reflect breaking internal change

illusionfield committed:
Revert "fix: resolve npm dependency installation issue in Meteor 2 package tests"

This reverts commit 122359fe99473e500b46f6d1b636e7843ef7cef4.

illusionfield committed:
typo

illusionfield committed:
Merge branch 'develop' into feature/remove-sass

illusionfield committed:
Merge branch 'feature/remove-sass' into develop

illusionfield committed:
v5.0.0 Beta 1

illusionfield commented:
@StorytellerCZ I'm done with the restructuring — I didn’t see a `release/5.0.0` branch, so I targeted `release-5` for now.  
We may want to clean up the release branches a bit later.  
I’ve also linked the issues that can be closed as part of this.

@jankapunkt `5.0.0-beta.1` is ready for release whenever you are.

illusionfield committed:
test suite

illusionfield commented:
Hi @jankapunkt @StorytellerCZ 👋

Just following up to check in — any updates on this?

`5.0.0-beta.1` is ready and waiting, and I’d love to get it out so the community can start testing and giving feedback.

Let me know if there's anything else you need from my side!

StorytellerCZ commented:
Published fourseven:scss@5.0.0-beta.1.

StorytellerCZ committed:
Published fourseven:scss@5.0.0-beta.1.

StorytellerCZ commented:
OK, I think we had enough in beta. I'm going to move to RC today to give more people a push to test it.

StorytellerCZ committed:
Published fourseven:scss@5.0.0-rc.1.

jankapunkt commented:
Please everyone keep testing and integrating, we will do a release on 1st December (unless someone detects bigger issues).

illusionfield committed:
update test

illusionfield committed:
switch automated test runner

- replace @zodern/mtest with @illusionfield/mtest and silence verbose output
- new fork is the maintained path, so we keep tests running reliably

illusionfield committed:
test deps fix: polyfill fetch for legacy Node runtimes

illusionfield committed:
update test:watch script to enable verbose output

StorytellerCZ commented on .github/workflows/test_suite.yml:
```diff
@@ -24,6 +24,8 @@ jobs:
           - meteor-version: 3.1.2
             node-version: 22
+          - meteor-version: 3.3.2
+            node-version: 24
```
Meteor 3.3.2 is still on Node 22.

illusionfield replied:
MeteorJS uses an embedded Node version internally, so the `node-version` defined here only affects the installation step.  
As far as I know, the installation should still work with the latest LTS version.  
Or would you recommend installing  an `nvm` environment using Node 22 first, and then proceeding with the Meteor installation from there?

jankapunkt commented:
A bit of delay but I am about to release during this week

illusionfield commented:
@jankapunkt
This will need to be rewritten.
https://github.com/Meteor-Community-Packages/meteor-scss/blob/589ab5a2b29401dd5010a47fda5dead73af37a73/README.md?plain=1#L8

jankapunkt commented:
@illusionfield do you have a prompt I can forward to copilot?

illusionfield commented:
@jankapunkt 
In README.md please update the Migration Guide link from:

  https://github.com/illusionfield/fourseven-scss/blob/master/MIGRATION-GUIDE-v5.md

to:

  https://github.com/Meteor-Community-Packages/meteor-scss/blob/master/MIGRATION-GUIDE-v5.md

It doesn’t need to point to my fork anymore, since it has been merged into the main repo.
